programs/system/reshare/reshare.asm

```diff
@@ -0,0 +1,546 @@
+        ; mov     [shared_i32], eax
+        mov     ebx, [icons32_image]
+        mov     esi, [ebx + Image.Data]
+        mov     edi, eax
+        mov     ecx, [size32]
+        shr     ecx, 2 ; / 4 to get size in dwords
+        cld
+        rep     movsd
+        ; now can img_destroy?
+    ; .else
+    ;     DEBUGF  DBG_ERR, "Daemon mode: no icons32 !\n"
+    .endif
+
+    .if [icons16_image] <> 0
+        mcall   SF_SYS_MISC, SSF_MEM_OPEN, shared_i16_name, [size16], SHM_CREATE + SHM_WRITE
+        ; mov     [shared_i16], eax
+        mov     ebx, [icons16_image]
+        mov     esi, [ebx + Image.Data]
+        mov     edi, eax
+        mov     ecx, [size16]
+        shr     ecx, 2 ; / 4 to get size in dwords
+        cld
+        rep     movsd
+
+        mcall   SF_SYS_MISC, SSF_MEM_OPEN, shared_i16w_name, [size16], SHM_CREATE + SHM_WRITE
+        mov     [shared_i16w], eax
+    ; .else
+    ;     DEBUGF  DBG_ERR, "Daemon mode: no icons16 !\n"
+    .endif
+
+    ; DEBUGF  DBG_INFO, "@reshare: shared successfully\n"
+    
+    mcall   SF_SET_EVENTS_MASK, 10000b ; set event mask EVM_DESKTOPBG
+.event_loop:
+    push    [sc.work]
+    mcall	SF_STYLE_SETTINGS, SSF_GET_COLORS, sc, sizeof.system_colors
+    pop     eax
+    .if eax <> [sc.work]
+        .if [icons16_image] <> 0
+            mov     ebx, [icons16_image]
+            mov     esi, [ebx + Image.Data]
+            mov     edi, [shared_i16w]
+            mov     ecx, [size16]
+            shr     ecx, 2 ; / 4 to get size in dwords
+            cld
+            rep     movsd
+            shl     ebx, 2 ; get size in bytes
+            stdcall replace_2cols, [shared_i16w], [size16], 0xffFFFfff, [sc.work], 0xffCACBD6, [sc.work_dark]
+        .endif
+    .endif
+    mcall   SF_WAIT_EVENT
+    cmp     eax, 5 ; redraw background event
+    je      .event_loop
+
+.exit:
+    mcall   SF_TERMINATE_PROCESS
+
+
+; data:
+include_debug_strings
+
+@imports:
+        library img, "libimg.obj"
+        import  img, img.destroy, "img_destroy", \
+                img.from_file,    "img_from_file"
+
+icons32_path    db "/SYS/ICONS32.PNG", 0
+icons16_path    db "/SYS/ICONS16.PNG", 0
+; pointers to Image structures
+icons32_image   dd 0
+icons16_image   dd 0
+; sizes of icons image data in bytes
+size32          dd 0
+size16          dd 0
+
+; shared_i32      dd 0
+shared_i32_name db "ICONS32", 0
+; shared_i16      dd 0
+shared_i16_name db "ICONS18", 0
+shared_i16w     dd 0
+shared_i16w_name db "ICONS18W", 0
+include         'chbox.inc'
+shared_chbox_name db "CHECKBOX", 0
+
+window_title    db "@RESHARE - A service that provides shared resources", 0
+window_top_text db "Each tab name corresponds to memory name that can be accessed by sysfunc 68.22. Now available:", 0
+
+active_tab      dd MASK_ACTIVE_ICONS32
+
+sc              system_colors
+
+align 16
+_image_end:
+
+thread_info     process_information ; NOTE: its kinda weird that struct for thread information called process_information
```

Comment:
Where is this structure defined?  Considering that threads and processes in KolibriOS do not differ in any way, there is no strangeness.  Perhaps we should make an alias to `process_information`.  For example `sf_thread_info_s`.

Reply:
`process_information` is defines in `programs/macros.inc`